### DISCUSSION TITLE
How to export and reload the model, and add image training on this basis

### DISCUSSION THREAD

**Discussion by GreenHandLJ** [deepjavalibrary/djl#1438] [Q&A]
I want use "ImageFolder" as "DataSet" to create and train a new Model then export it,
when I have some new pictures, I can reload this model and train these new pictures on this model

I use different picture to predict ,but all results are the same.
Where can I find the demo

 after directory:
models
    shoeclassifier-0010.params
    synset.txt

 result:
[
	class: "大蒜", probability: 1.00000
	class: "包菜", probability: 0.0e+00
]

Training:
public final class Training {

    private static final int BATCH_SIZE = 32;
    private static final int EPOCHS = 10;
    public static final int IMAGE_HEIGHT = 416;
    public static final int IMAGE_WIDTH = 416;

    public static void main(String[] args) throws IOException, TranslateException, MalformedModelException {
        Path modelDir = Models.MODEL_PATH;
        ImageFolder dataset = initDataset(new File("E:\\ai\\ai\\data\\training\\min1").toPath());//size 2
        ImageFolder datasetValiddate = initDataset(new File("E:\\ai\\ai\\data\\validate\\min1").toPath());//size 2
//        ImageFolder dataset = initDataset(new File("E:\\ai\\ai\\data\\training\\min2").toPath());//size 1
//        ImageFolder datasetValiddate = initDataset(new File("E:\\ai\\ai\\data\\validate\\min2").toPath());//size 1
        TrainingConfig config = setupTrainingConfig(Loss.softmaxCrossEntropyLoss());
        Model model = Model.newInstance(Models.MODEL_NAME);
        Block resNet50 =
                ResNetV1.builder() // construct the network
                        .setImageShape(new Shape(3, IMAGE_HEIGHT, IMAGE_WIDTH))
                        .setNumLayers(50)
                        .setOutSize(3)
                        .build();
        model.setBlock(resNet50);
        //load exits model
        if (modelDir.toFile().exists()) {
            model.load(modelDir, Models.MODEL_NAME);
        }

        Trainer trainer = model.newTrainer(config);
        trainer.setMetrics(new Metrics());
        Shape inputShape = new Shape(1, 3, IMAGE_HEIGHT, IMAGE_WIDTH);
        trainer.initialize(inputShape);

        //train
        EasyTrain.fit(trainer, EPOCHS, dataset, datasetValiddate);

        TrainingResult result = trainer.getTrainingResult();
        model.setProperty("Epoch", String.valueOf(EPOCHS));
        model.setProperty(
                "Accuracy", String.format("%.5f", result.getValidateEvaluation("Accuracy")));
        model.setProperty("Loss", String.format("%.5f", result.getValidateLoss()));

        //save model
        model.save(modelDir, Models.MODEL_NAME);
        Models.saveSynset(modelDir, dataset.getSynset());
    }

    private static ImageFolder initDataset(Path datasetPath)
            throws IOException, TranslateException {
        ImageFolder dataset =
                ImageFolder.builder()
                        // retrieve the data
                        .setRepositoryPath(datasetPath)
                        .optMaxDepth(10)
                        .addTransform(new Resize(IMAGE_WIDTH, IMAGE_HEIGHT))
                        .addTransform(new ToTensor())
                        // random sampling; don't process the data in order
                        .setSampling(BATCH_SIZE, true)
                        .build();

        dataset.prepare();
        return dataset;
    }

    private static TrainingConfig setupTrainingConfig(Loss loss) {
        return new DefaultTrainingConfig(loss)
                .addEvaluator(new Accuracy())
                .addTrainingListeners(TrainingListener.Defaults.logging());
    }
}

predict:
public class Inference {

    public static void main(String[] args) throws ModelException, TranslateException, IOException {
        Image img = ImageFactory.getInstance().fromFile(new File("E:\\ai\\ai\\data\\training\\min1\\包菜\\1包菜.jpg").toPath());

        try (Model model = Models.getModel()) { // empty model instance
            // load the model
            model.load(Models.MODEL_PATH, Models.MODEL_NAME);

            Translator<Image, Classifications> translator =
                    ImageClassificationTranslator.builder()
                            .addTransform(new Resize(Models.IMAGE_WIDTH, Models.IMAGE_HEIGHT))
                            .addTransform(new ToTensor())
                            .optApplySoftmax(true)
                            .build();

            try (Predictor<Image, Classifications> predictor = model.newPredictor(translator)) {
                Classifications predictResult = predictor.predict(img);
                System.out.println(predictResult);
            }
        }
    }
}

**Comment by zachgk** [comment]
Most of your code seems right, but I am not quite sure what exactly your question is. However, I notice that your dataset has 2 elements and you need to have much more than that. As a comparison, MNIST which is a pretty basic image dataset has 60,000 training images and 10,000 test images. 

To retrain a model, you can load it with `model.load(...)` and then run the usual training process. Make sure to skip the `trainer.initialize(...)` which puts random values into the model.
